tests/smcp_gold_panda.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"maistra/util"
+	"testing"
+	"time"
+
+	"istio.io/pkg/log"
+)
+
+func cleanuGoldPandaSMCP(namespace string) {
+	log.Info("# Cleanup ...")
+	util.KubeDelete(namespace, goldPandaSMCP, kubeconfig)
+	time.Sleep(time.Duration(waitTime*8) * time.Second)
+	// avoid namespace recreation for downstream service account settings
+}
+
+func TestGoldPandaSMCP(t *testing.T) {
+	defer cleanuGoldPandaSMCP("istio-gold-panda")
+
+	t.Run("Operator_test_smcp_gold_panda", func(t *testing.T) {
+
+		defer recoverPanic(t)
+
+		// // create a smcp with nil redudant policies field for Jaeger - See MAISTRA-1983
+		// util.CreateOCPNamespace("istio-gold-panda", kubeconfig)
+		// log.Info("Update SMCP with gold panda smcp")
+		// if err := util.KubeApply("istio-gold-panda", goldPandaSMCP, kubeconfig); err != nil {
+		// 	t.Errorf("Failed to deploy SMCP with goldpanda")
+		// 	log.Errorf("Failed to deploy SMCP with goldpanda")
+		// }
+		util.Shell(`kubectl patch -n %s smcp/%s --type merge -p '{"spec":{"techPreview":{"global":{"pathNormalization":"DECODE_AND_MERGE_SLASHES"}}}}'`, meshNamespace, smcpName)
```

Comment:
Did we want tests for the other options as well? These were separated in the release as each level constitutes breaking changes for some users.

Reply:
@brian-avery that's a good suggestion. We can add similar sub test cases in that go test script later.  The MAISTRA-2354 mentioned DECODE_AND_MERGE_SLASHES option. So that should be enough for this PR. following PRs can be added with NONE, BASE or MERGE_SLASHES options in sub test cases. What do you think @gbaufake  ?